backends/redis.go

```diff
@@ -0,0 +1,118 @@
+package backends
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+
+	"github.com/go-redis/redis"
+)
+
+type RedisBackend struct {
+	client *redis.Client
+}
+
+func NewRedisBackend(connString string) (*RedisBackend, error) {
+	u, _ := url.Parse(connString)
+	dbParam := u.Query().Get("database")
+
+	// Default to DB 0
+	var db int
+	if dbParam != "" {
+		fmt.Print(dbParam)
+		db, _ = strconv.Atoi(dbParam)
```

Comment:
What if someone puts a "💩" here? Probably should do something with this error instead of ignoring it.